### PR TITLE
Fix not being able to read in full bright areas

### LINKED ISSUE #73830: Thunderdome is "too dark in here to read" on certain tiles after being reset
Reporting client version: 514.1589

## Round ID:
[201356](https://scrubby.melonmesa.com/round/201356)

## Testmerges:
- [[NO GBP] Fixes some major (and minor) issues with the birthday station trait.](https://www.github.com/tgstation/tgstation/pull/73813)

## Reproduction:

Reset Thunderdome via Secrets panel, then spawn in with a health analyzer.  Analyze your vitals and get "It's too dark in here to read!" error only on certain tiles.  I did some testing...below seems to be a consistent pattern seen across two shifts (trust the process, ignore the insane execution).  All tiles with cats are "too dark to read'.

image

### LINKED ISSUE #74107: The abductor ship is too dark to read the dissection guide.
Reporting client version: 514.1589

## Round ID:
[202089](https://scrubby.melonmesa.com/round/202089)

## Testmerges:
- [Nanotrasen Budget Programme - Mothball Edition [BIRDSHOT STATION]](https://www.github.com/tgstation/tgstation/pull/73502)

## Reproduction:
The dissection guide for abductors cannot be read because there is no light source in the abductor's ship. All the turfs are lit up, but the lack of light source prevents the reading.
1. Become an abductor.
2. Pick up and try to read the guide.
3. Fail to read.

### PR DESCRIPTION
**Pull request by timothymtorres**
## About The Pull Request
Fixes #73830
Fixes #74107

There seems to be a problem when the map template reloads for full bright areas, but only for thunderdome specifically.  Very strange and my fix feels like a band-aid but it's a niche bug report.

Also deleted an area that was not used anywhere in the codebase.

## Why It's Good For The Game
Silly bug go bye bye.

## Changelog
:cl:
fix: Fix not being able to read in full bright areas
/:cl:

### CODE REVIEW

File: code/modules/mob/mob.dm [hunk 1]
```diff
@@ -1232,7 +1232,15 @@
 **/
 /mob/proc/has_light_nearby(light_amount = LIGHTING_TILE_IS_DARK)
 	var/turf/mob_location = get_turf(src)
-	return mob_location.get_lumcount() > light_amount
+	var/area/mob_area = get_area(src)
+
+	if(mob_location.get_lumcount() > light_amount)
+		return TRUE
+	else if(!mob_area.static_lighting)
+		return TRUE
```
Comment (san7890) [inline code/modules/mob/mob.dm:1240]:
```suggestion
	if(!mob_area.static_lighting || mob_location.get_lumcount() > light_amount)
		return TRUE
```

Why not do it like this?

**Review by san7890 — CHANGES_REQUESTED** [review]

File: code/modules/mob/mob.dm (hunk 1 above)
Comment (LemonInTheDark) [reply code/modules/mob/mob.dm:1240]: Yes it means the opposite of what you think

Comment (timothymtorres) [reply code/modules/mob/mob.dm:1240]: I'm aware of ternary operators, but you should be careful with nesting them too deeply or if it's involving complex code logic.

Comment (LemonInTheDark) [reply code/modules/mob/mob.dm:1240]: that isn't a ternary operator

**Comment by Sealed101** [comment]
does this fix #74107 ?

**Comment by timothymtorres** [comment]
> does this fix #74107 ?

Yes it should.

**Pull request merged**